File: C++/014. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int size = strs.size();
        string result = "";
        if(size==0)
        {
            return result;
        }
        for(int j=0;j<strs[0].length();j++)
        {
            for(int i=0;i<size;i++)
            {
                if(strs[i][j]!=strs[0][j])
                {
                    return result;
                }
            }
            result += strs[0][j];
        }
        return result;
    }
};

File: python/014. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0: return ""
        if len(strs) == 1: return strs[0]
        n = sys.maxsize
        for s in strs:
            n = min(n, len(s))
        res = ""
        for i in range(n):
            cur = strs[0][i]
            for s in strs:
                if s[i] != cur: return res
            res += cur
        return res
